src/Npgsql/NpgsqlTransaction.cs

```diff
@@ -204,14 +201,33 @@ public void Save(string name)
             using (_connector.StartUserAction())
             {
                 Log.Debug($"Creating savepoint {name}", _connector.Id);
-                _connector.ExecuteInternalCommand($"SAVEPOINT {name}");
+                await _connector.ExecuteInternalCommand($"SAVEPOINT {name}", async);
             }
         }
 
         /// <summary>
-        /// Rolls back a transaction from a pending savepoint state.
+        /// Creates a transaction save point.
         /// </summary>
-        public void Rollback(string name)
+        public void Save(string name) => Save(name, false).GetAwaiter().GetResult();
+
+        /// <summary>
+        /// Creates a transaction save point.
+        /// </summary>
+        [PublicAPI]
+        public Task SaveAsync(string name, CancellationToken cancellationToken)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
```

Comment:
Note that this isn't the right pattern - a cancelled Task should be returned rather than throwing. However, we don't do this correctly in some other async methods, I've opened #2417 to track this separately.